src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -851,7 +851,7 @@ internal HttpListenerContext HandleAuthentication(RequestContextBase memoryBlob,
                             NetEventSource.Info(this, $"authenticationScheme: {authenticationScheme}");
                         }
                         SendError(requestId, HttpStatusCode.InternalServerError, null);
-                        httpContext.Close();
+                        FreeContext(ref httpContext, memoryBlob);
```

Comment:
We're now calling DetachBlob where we weren't before? That's desired? Was this a leak previously?

Reply:
We call `DetachBlob` function to save the memoryBlob from disposing if it's never given to users, so that we can reuse the memoryBlob object (instead of initializing a new one, if new blob requires same size), to increase performance.

In this case, HandleAuthentication failed at `AuthenticationSchemeSelectorDelegate` (at the very beginning), we need to dispose the HttpListenerContext, but still want to keep the memoryBlob (in cases we could reuse it).

Reply:
So it was previously a bug that this change is fixing?

Reply:
Yes, we should keep the memoryBlob (not dispose it right away) so that it could be reused.

Reply:
Ok, thanks.